File: launcher/ui/IndexSche.py
# import PySide2.QtCore
# import PySide2.QtWidgets
import sys

from PySide2.QtWidgets import QWidget, QVBoxLayout, QRadioButton, QGroupBox, QFormLayout, QLabel, QLineEdit, \
    QPushButton, QHBoxLayout, QApplication, QCheckBox, QDialogButtonBox


class IndexSche(QWidget):
    def __init__(self):
        super().__init__()
        self._init_ui()
        self._init_conn()

    def _init_ui(self):
        # self._btn_cron=QRadioButton(self.tr("Cron"))
        # self._btn_realtime=QRadioButton(self.tr("RealTime"))
        vlayout = QVBoxLayout()
        self.setLayout(vlayout)

        _group_cron = QGroupBox(self.tr("Cron"))
        vlayout.addWidget(_group_cron)

        _formLayout_Cron = QFormLayout()
        _group_cron.setLayout(_formLayout_Cron)

        _formLayout_Cron.addWidget(QLabel("xxxx"))
        self._label_cron_week = QLineEdit()
        self._label_cron_hour = QLineEdit()
        self._label_cron_minu = QLineEdit()
        _formLayout_Cron.addRow(self.tr("星期日"), self._label_cron_week)
        _formLayout_Cron.addRow(self.tr("小时"), self._label_cron_hour)
        _formLayout_Cron.addRow(self.tr("分钟"), self._label_cron_minu)

        self._btn_enable = QPushButton(self.tr("Enable"))
        self._btn_disable = QPushButton(self.tr("Disable"))
        # self._btn_cancel = QPushButton(self.tr("Cancel"))
        hlayout = QHBoxLayout()
        hlayout.addWidget(self._btn_enable)
        hlayout.addWidget(self._btn_disable)
        # hlayout.addWidget(self._btn_cancel)
        # self._btn_cron=QDialogButtonBox(QDialogButtonBox)

        _formLayout_Cron.addWidget(QLabel("ddd"))
        _formLayout_Cron.addRow(hlayout)

        _group_realtime = QGroupBox(self.tr("RealTime"))
        vlayout.addWidget(_group_realtime)

        formLayout=QFormLayout()
        _group_realtime.setLayout(formLayout)

        formLayout.addWidget(QLabel(self.tr("xxx")))

        self._cbk_startOnStart=QCheckBox()
        self._cbk_startNow=QCheckBox()
        self._cbk_startNow.setEnabled(False)
        formLayout.addRow(self._cbk_startOnStart,QLabel(self.tr("Start on System Start")))
        formLayout.addRow(self._cbk_startNow,QLabel(self.tr("Start Now")))



        pass

    def _init_conn(self):
        self._btn_enable.clicked.connect(self._cron_enable)
        self._btn_disable.clicked.connect(self._cron_disable)
        self._cbk_startOnStart.toggled.connect(self._cbk_onstart_toggled)
        self._cbk_startNow.toggled.connect(self._cbk_start_now_toggled)
        pass
    def _cbk_onstart_toggled(self,checked):
        if checked:
            self._cbk_startNow.setEnabled(True)
        ## TODO imple


        pass
    def _cbk_start_now_toggled(self,checked):
        pass
    def _cron_enable(self):
        pass
    def _cron_disable(self):
        pass


if __name__ == "__main__":
    app = QApplication()
    index = IndexSche()
    index.show()

    sys.exit(app.exec_())

File: src/indexsche.cpp
#include "indexsche.h"

#include <QFormLayout>
#include <QGroupBox>
#include <QLabel>
#include <QVBoxLayout>

IndexSche::IndexSche(QWidget *parent) : QWidget(parent)
{
    this->lbCronHour=new QLineEdit();
    this->lbCronMinu=new QLineEdit();
    this->lbCronWeek=new QLineEdit();
    this->btnDisable=new QPushButton(tr("Disable"));
    this->btnEnable=new QPushButton(tr("Enable"));
    this->ckbStartNew=new QCheckBox();
    this->ckbStartOnStart=new QCheckBox();

    init_ui();
    init_conn();
}

void IndexSche::init_ui()
{
    auto vlayout=new QVBoxLayout();
    this->setLayout(vlayout);
    auto groupCron=new QGroupBox(tr("Cron"));
    vlayout->addWidget(groupCron);
    auto formLayoutCron=new QFormLayout();
    groupCron->setLayout(formLayoutCron);

    formLayoutCron->addWidget(new QLabel("xxx"));
    formLayoutCron->addRow(tr("星期"),this->lbCronWeek);
    formLayoutCron->addRow(tr("小时"),this->lbCronHour);
    formLayoutCron->addRow(tr("分钟"),this->lbCronMinu);

    auto hlayout=new QHBoxLayout();
    hlayout->addWidget(this->btnEnable);
    hlayout->addWidget(this->btnDisable);
    formLayoutCron->addWidget(new QLabel("xxx"));
    formLayoutCron->addRow(hlayout);

    auto groupRealtim=new QGroupBox(tr("RealTime"));
    vlayout->addWidget(groupRealtim);

    auto formLayout=new QFormLayout();
    groupRealtim->setLayout(formLayout);
    formLayout->addWidget(new QLabel(tr("aaa")));


    this->ckbStartNew->setEnabled(false);
    formLayout->addRow(this->ckbStartOnStart,new QLabel("Start On System Start"));
    formLayout->addRow(this->ckbStartNew,new QLabel(tr("Start Now")));

}

void IndexSche::init_conn()
{
    connect(this->btnEnable,&QPushButton::clicked,[](){
        //TODO
    });
    connect(this->btnDisable,&QPushButton::clicked,[](){
        //TODO
    });
    connect(this->ckbStartNew,&QCheckBox::toggled,[](){

    });
    connect(this->ckbStartOnStart,&QCheckBox::toggled,[this](bool checked){
        if(checked){
            this->ckbStartNew->setEnabled(true);
        }

    });

}
